support-compat version problem causes build failure

## 🐛 Bug Report

<!--- Summary description of the bug --->
After adding the plugin to pubspec.yaml, the project fails to build:
`FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:preDebugBuild'.
> Android dependency 'com.android.support:support-compat' has different version for the compile (27.0.2) and runtime (27.1.1) classpath. You should manually set the same version via DependencyResolution
`
### Expected behavior
Successful build. 

### Reproduction steps
Build a project with these plugins in pubspec.yaml:
```
name: flutter_app
description: A new Flutter application.

dependencies:
  http: "^0.11.3+16"
  flutter:
    sdk: flutter

  cupertino_icons: ^0.1.0
  cached_network_image: "^0.4.0"
  firebase_auth: ^0.5.20
  intl: ^0.15.6
  flutter_launcher_icons: "^0.6.1"
  font_awesome_flutter: ^8.0.1
  flutter_facebook_login: ^1.1.1
  google_sign_in: ^3.0.5
  shared_preferences: ^0.4.3
  location: ^1.4.1
  permission_handler: '^2.0.0'

flutter_icons:
  image_path: "icon/u.png"
  android: trued
  ios: true

dev_dependencies:
  flutter_test:
    sdk: flutter

flutter:
  uses-material-design: true

```

### Configuration

**Version:** 1.x and 2.x

**Platform:** 
- [ ] :iphone: iOS
- [x ] :robot: Android

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -40,7 +40,8 @@ android {
 }
 
 dependencies {
+    api "com.android.support:support-compat:27.1.1"
+
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
     implementation "com.google.code.gson:gson:2.8.5"
-    implementation "com.android.support:support-compat:27.1.1"
 }

